scripts/bash/run_enceladus.sh

```diff
@@ -254,11 +287,16 @@ add_to_cmd_line "--master" ${MASTER}
 add_to_cmd_line "--deploy-mode" ${DEPLOY_MODE}
 add_to_cmd_line "--num-executors" ${NUM_EXECUTORS}
 add_to_cmd_line "--executor-memory" ${EXECUTOR_MEMORY}
+add_to_cmd_line "--executor-cores" ${EXECUTOR_CORES}
 add_to_cmd_line "--driver-cores" ${DRIVER_CORES}
 add_to_cmd_line "--driver-memory" ${DRIVER_MEMORY}
 
+# Adding Spark config options
+add_spark_conf_cmd "spark.executor.memoryOverhead" ${CONF_SPARK_EXECUTOR_MEMORY_OVERHEAD}
+add_spark_conf_cmd "spark.memory.fraction" ${CONF_SPARK_MEMORY_FRACTION}
+
 # Adding JVM configuration, entry point class name and the jar file
-CMD_LINE="${CMD_LINE} --conf \"${CONF} ${ADDITIONAL_SPARK_CONF}\" --class ${CLASS} ${JAR}"
+CMD_LINE="${CMD_LINE} ${ADDITIONAL_SPARK_CONF} --conf \"${JVM_CONF} ${ADDITIONAL_JVM_CONF}\" --class ${CLASS} ${JAR}"
```

Comment:
This seems a bit weird to me. On one hand, you have `ADDITIONAL_SPARK_CONF` where you ask for spark config options and then you have the lines `295` and `296` where you ask for specific spark config options. Can these be written so that they overwrite themselves in case of both of them being specified? Maybe add a check in `add_spark_conf_cmd` with a regex that would overwrite the value?

Reply:
* Specific Spark options are used to fine-tune each individual job (e.g. specify memory overhead for super big files).
* Additional Spark configuration options are used for additional environment configuration, e.g. to use scripts from containers (`spark.driver.host`, for example)

Reply:
I understand that, but I think you are forcing users to understand the innards of the script, otherwise, they could specify the `--conf spark.executor.memoryOverhead` twice.  I am suggesting we "handle" that.

Reply:
I see now, thanks. Probably we need to handle it so that options provided by the command line override ones specified in `enceladus_env.sh`. I'm going to check if it behaves like this in this implementation. 

In this implementation command line provided options go after the ones specified in `enceladus_env.sh`. I'm going to check which option becomes effective.

Reply:
Thanks for spotting this! These are valid points. I confirm that the current logic is reverted.

When you specify the same config key twice with different values, Spark sets the second (latter) value to be the effective configuration value.
E.g. for
```
spark-submit --conf spark.executor.memoryOverhead=500M --conf spark.executor.memoryOverhead=600M
```
Spark going to set
```
spark.executor.memoryOverhead=600M
```
Will change the script so that the command line provided options take precedence over environment defaults.